constants/battle.s

```diff
@@ -53,3 +53,29 @@
 	enum SP_DEFENSE
 	enum ACCURACY
 	enum EVASION
+
+@ battle_data_t
+.equiv Battle_Species,	    0x00
+.equiv Battle_Stats,        0x02 @ atk/def/spe/spatk/spdef
+.equiv Battle_Move_IDs,     0x0C @ four attacks
+.equiv Battle_IVs,          0x14
+.equiv Battle_Stat_Buffs,   0x18 @ hp/atk/def/spe/spatk/spdef
+.equiv Battle_Ability_ID,   0x20
+.equiv Battle_Type_1,       0x21
+.equiv Battle_Type_2,       0x22
+.equiv Battle_unk,          0x23 @undefined
+.equiv Battle_PP,           0x24
+.equiv Battle_Current_HP,   0x28
+.equiv Battle_Level,        0x2A
+.equiv Battle_Happiness,    0x2B
+.equiv Battle_Max_HP,       0x2C
+.equiv Battle_Held_Item,    0x2E
+.equiv Battle_Name,         0x30
+.equiv Battle_3B,           0x3B
+.equiv Battle_Trainer_Name, 0x3C
+.equiv Battle_padding,      0x44
+.equiv Battle_PID,          0x48
+.equiv Battle_Status1,      0x4C
+.equiv Battle_Status2,      0x50
+.equiv Battle_OTID,         0x54
+.equiv Battle_End,          0x58
```

Comment:
This should probably be a struct.

```c
struct BattlePokemon {
 	u16 species;
 	u16 attack, defense, speed, sp_attack, sp_defense;
 	u16 move1, move2, move3, move4;
 	u8 base_hp, base_attack, base_defense, base_speed, base_sp_attack, base_sp_defense;
 	u8 attack_level, defense_level, speed_level, sp_attack_level, sp_defense_level;
 	u16 ability;
 	u8 type1, type2;
 	u8 unknown;
 	u8 pp1, pp2, pp3, pp4;
 	u16 hp;
 	u8 level;
 	u8 friendship;
 	u16 max_hp;
 	u16 item;
 	char[11] nickname;
 	u8 unknown2;
 	char[8] ot_name;
 	int unknown3;
 	int id;
 	int status;
 	int substatus;
 	int ot_id;
}
```
This is not usable in asm on its own. You could use `.struct` to make an asm counterpart.

Example: https://github.com/pret/pokeruby/blob/849b131c1edd351d620b0d335d67904e4689d1fe/constants/m4a_constants.s#L10-L17
```asm
	.macro struct_begin
	.struct 0
	.endm

	.macro struct_field name, size
\name:
	.struct \name + \size
	.endm
```
```asm
	struct_begin
	struct_field BattlePokemon_species, 2
	struct_field BattlePokemon_attack, 2
	...
```